clients/python/agentic-sandbox-client/agentic_sandbox/sandbox_client.py

```diff
@@ -113,37 +121,50 @@ def _wait_for_sandbox_ready(self):
             field_selector=f"metadata.name={self.claim_name}",
             timeout_seconds=self.sandbox_ready_timeout
         ):
-            sandbox_object = event['object']
-            status = sandbox_object.get('status', {})
-            conditions = status.get('conditions', [])
-            is_ready = False
-            for cond in conditions:
-                if cond.get('type') == 'Ready' and cond.get('status') == 'True':
-                    is_ready = True
-                    break
+            if event["type"] in ["ADDED", "MODIFIED"]:
+                sandbox_object = event['object']
+                status = sandbox_object.get('status', {})
+                conditions = status.get('conditions', [])
+                is_ready = False
+                for cond in conditions:
+                    if cond.get('type') == 'Ready' and cond.get('status') == 'True':
+                        is_ready = True
+                        break
 
-            if is_ready:
-                self.sandbox_name = sandbox_object['metadata']['name']
-                w.stop()
-                logging.info(f"Sandbox {self.sandbox_name} is ready.")
-                break
+                if is_ready:
```

Comment:
This new logic for discovering the pod name is critical for warm pool functionality but is not covered by automated tests. Please add unit tests to `test_client.py` to validate this behavior and prevent future regressions. The tests should cover cases where the annotation is present and where it is absent.

Reply:
This is tracked under #151.